cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -4183,10 +4183,48 @@ TEST(TestArrowReadDeltaEncoding, DeltaBinaryPacked) {
 
   ::arrow::AssertTablesEqual(*table, *expect_table);
 }
+
+TEST(TestArrowReadDeltaEncoding, DeltaByteArray) {
+  auto file = test::get_data_file("delta_byte_array.parquet");
+  auto expect_file = test::get_data_file("delta_byte_array_expect.csv");
+  auto pool = ::arrow::default_memory_pool();
+  std::unique_ptr<FileReader> parquet_reader;
+  std::shared_ptr<::arrow::Table> parquet_table;
+  ASSERT_OK(
+      FileReader::Make(pool, ParquetFileReader::OpenFile(file, false), &parquet_reader));
+  ASSERT_OK(parquet_reader->ReadTable(&parquet_table));
+  ASSERT_OK_AND_ASSIGN(auto actural_table, parquet_table->CombineChunks());
+
+  ASSERT_OK_AND_ASSIGN(auto input_file, ::arrow::io::ReadableFile::Open(expect_file));
+  auto convert_options = ::arrow::csv::ConvertOptions::Defaults();
+  std::array<std::string, 12> column_names = {
```

Comment:
Why 12? You seem to have 9 of them :-)
More idiomatically, just use `std::vector<std::string>`?